test/extensions/KHR/SPV_KHR_untyped_pointers/infinite-phi.ll

```diff
@@ -0,0 +1,55 @@
+; RUN: llvm-as %s -o %t.bc
+; RUN: llvm-spirv %t.bc --spirv-ext=+SPV_KHR_untyped_pointers -o %t.spv
+; TODO: enable back once spirv-tools are updated.
+; R/UN: spirv-val %t.spv
+; RUN: llvm-spirv %t.bc --spirv-ext=+SPV_KHR_untyped_pointers -spirv-text -o %t.spt
+; RUN: FileCheck < %t.spt %s --check-prefix=CHECK-SPIRV
+
+; RUN: llvm-spirv -r %t.spv -o %t.rev.bc
+; RUN: llvm-dis < %t.rev.bc
+; RUN: llvm-dis %t.rev.bc -o %t.rev.ll
+; RUN: FileCheck < %t.rev.ll %s --check-prefix=CHECK-LLVM
+
+target datalayout = "e-p:32:32-i64:64-v16:16-v24:32-v32:32-v48:64-v96:128-v192:256-v256:256-v512:512-v1024:1024"
+target triple = "spir-unknown-unknown"
+
+; CHECK-SPIRV: TypeInt [[#INT:]] 32 0
+; CHECK-SPIRV: TypeUntypedPointerKHR [[#PTRTY:]] 7
+; CHECK-SPIRV: TypeFloat [[#FLOAT:]] 32
+
+; CHECK-SPIRV: UntypedVariableKHR [[#PTRTY]] [[#IPTR:]] 7 [[#INT]]
+; CHECK-SPIRV: UntypedVariableKHR [[#PTRTY]] [[#FPTR:]] 7 [[#FLOAT]]
+; CHECK-SPIRV: Bitcast [[#PTRTY]] [[#IPTRI:]] [[#IPTR]]
```

Comment:
This bitcast feels redundant. Guess it's being generated by type scavenger, right? It might be done in a separate PR, but probably we can get rid of it by either modifying the scavenger or by checking the types during bitcast generation.

Reply:
You're right, the bitcast appeared because of the type scavenger. This happens before any translation, so for now I'm not sure how to avoid this issue. I'll add TODO to the test